### ISSUE TITLE
xterm-termite: unknown terminal type

### ISSUE THREAD

**Issue by varac** [mobile-shell/mosh#959]
I'm using the [termite terminal](https://github.com/thestinger/termite) and want to start a new instance with mosh running in it.

I'm starting it like this:

```
termite --hold -e 'mosh some.server'
(termite:10041): GLib-WARNING **: GChildWatchSource: Exit status of a child process was requested but ECHILD was received by waitpid(). Most likely the process is ignoring SIGCHLD, or some other thread is invoking waitpid() with a nonpositive first argument; either behavior can break applications that use g_child_watch_add()/g_spawn_sync() either directly or indirectly.
```

And the new terminal shows this mosh error:
```
'xterm-termite': unknown terminal type.
Died at /usr/bin/mosh line 311.
```

mosh shoul gracefully catch an unknown terminal type and not fail imo.

**Comment by cgull** [comment]
The Termite error message is an issue with Termite and is unrelated to Mosh, as can be seen by running `termite --hold -e false`.

As for Mosh, it does need to query terminfo to detect support of certain terminal features.  The error you're getting is apparently because Termite's terminal type `xterm-termite` is not configured in your system's terminfo database.

You can work around this by setting the environment variable `TERM=xterm-256color` before starting Mosh. Unfortunately, Termite doesn't seem to have a configuration option for this.  I don't know why Termite maintains its own terminfo entry, instead of using an existing one.

What improvement are you looking for?  If you're looking for Mosh to be able to handle an unknown terminal type and continue running, it won't do that currently (but we could have it simply decide that the terminfo capabilities it wants are unavailable-- they affect Mosh behavior in fairly minor ways and we could run without them).  If you want a better error message, what would you suggest?

**Comment by varac** [comment]
I'd be most interested in a fallback mechanism for mosh so it doesn't bail on an unknown terminal type.

**Comment by keithw** [comment]
Do you not have the same problem when running *any* termcap/terminfo application on the client (e.g. screen, tmux, pine, vim)? If so, this doesn't really seem like Mosh's problem...

**Comment by varac** [comment]
@keithw oh you're right, I just tried `termite --hold -e 'tmux'` which fails in a similar way.
So I'll take this over to termite!
